File: Leetcode/All/0001_two_sum/solution.ts
function twoSum(nums: number[], target: number): number[] {
    const map: Map<number, number> = new Map();
    
    for (let i = 0; i < nums.length; i++) {
        const num: number = nums[i];
        const complement: number = target - num;
        
        if (map.has(complement)) {
            return [map.get(complement) as number, i];
        }
        
        map.set(num, i);
    }
    
    return [];
}

File: Leetcode/All/0001_two_sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            int complement = target - num;

            if (map.find(complement) != map.end()) {
                return { map[complement], i };
            }

            map[num] = i;
        }

        return {};
    }
};
